### ISSUE TITLE
[P2] Cube-to-simplex clipping excludes near-vertex portfolios and breaks valid interior round trips

### ISSUE THREAD

**Issue by microprediction** [microprediction/humpday#358]
Reviewed at `b67729116580c5bf48899dd08c1b86690a6e1e9b` (main, v0.24.1).

The README advertises a smooth map onto the simplex. The actual [transform](https://github.com/microprediction/humpday/blob/b67729116580c5bf48899dd08c1b86690a6e1e9b/humpday/transforms/thurstone_transform.py#L33) clips every cube coordinate to [1e-10, 1-1e-10], then applies probit and an unscaled softmax with one reference logit fixed at zero.

For a two-component simplex this restricts weights to approximately [0.00172407, 0.99827593]. That excludes a material set of valid interior portfolios, not only exact boundary points:

```python
from humpday.transforms.cubetosimplex import cube_to_simplex, simplex_to_cube
p = [.9999, .0001]
u = simplex_to_cube(p)
print(u)                       # [0.0]
print(cube_to_simplex(u))       # [0.998275928..., 0.001724071...]
print(cube_to_simplex([0.0]))   # same clipped endpoint
```

An objective whose optimum needs a component below roughly 0.17% cannot reach it through this 1D transform, regardless of the optimizer budget. Comments claiming corners are reachable and any positive `STD_L` produces a bijection also overstate the implemented floating-point mapping.

Define and implement the intended numerical domain: provide a mapping/conditioning strategy that can represent the intended concentrated allocations, use stable operations, and specify treatment of zero/boundary weights. If truncation is intentional, expose/document its actual attainable region and avoid describing the implementation as onto the full simplex. The illustrated tunable parameter should correspond to a supported Python API or be clearly distinguished.

Acceptance: high-concentration interior allocations and representative higher-dimensional cases; documented boundary behavior; validation of invalid probability vectors; round-trip/optimization tests that would catch the excluded near-vertex region. No floating-point implementation needs to promise an exact mathematical bijection at all boundaries.
